plugin/RecordDotPreprocessor.hs

```diff
@@ -228,3 +231,23 @@ adjacentBy i (L (srcSpanEnd -> RealSrcLoc a) _) (L (srcSpanStart -> RealSrcLoc b
     srcLocLine a == srcLocLine b &&
     srcLocCol a + i == srcLocCol b
 adjacentBy _ _ _ = False
+
+
+makeEqQualTy :: HsType GhcPs -> (HsType GhcPs -> HsType GhcPs) -> HsType GhcPs
```

Comment:
Can you include an example - e.g. `Given t and f, it produces `t ~ f aplg` or whatever (its super hard to read these syntax productions in the abstract)